### ISSUE TITLE
ExtensionLoader only collect exceptions but do not throws when loadResource

### ISSUE THREAD

**Issue by fengbaicanhe** [apache/dubbo#3575]
**Dubbo version: master**
**JDK version: 1.8**

**Reproduce**
SPI config file: 

META-INF/dubbo/internal/com.limore.ExtensionConflict
`impl1=com.limore.impl.ExtensionConflictImpl1`

META-INF/dubbo/com.limore.ExtensionConflict
`impl1=com.limore.impl.ExtensionConflictImpl2`

**Note: the SPI file in different directory**

then use ExtensionLoader load ExtensionConflict 
```
ExtensionLoader.getExtensionLoader(ExtensionConflict.class).getExtension("impl1") // return  ExtensionConflictImpl1
```

**result: no exception occured, ExtensionConflictImpl2 loaded by class loader, but returned ExtensionConflictImpl1**

**Expectation: throw 'Duplicate extension' exception**

**The Reason**
[ExtensionLoader#L764](https://github.com/apache/incubator-dubbo/blob/master/dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java#L764) throw 'duplicate extension' exception, and [ExtensionLoader#LL706](https://github.com/apache/incubator-dubbo/blob/master/dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java#L706) collect the exception, but there is no any code process the collected exceptions. reference [ExtensionLoader#L661](https://github.com/apache/incubator-dubbo/blob/master/dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java#L661) and [ExtensionLoader#L628](https://github.com/apache/incubator-dubbo/blob/master/dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java#L628) method.

**Comment by pinxiong** [comment]
Let me have a look

**Comment by pinxiong** [comment]
@fengbaicanhe Very sorry to solve the problem you feedback so late.

This issue cannot reproduce on `master` branch you have mentioned 

```java
private void saveInExtensionClass(Map<String, Class<?>> extensionClasses, Class<?> clazz, String name, boolean overridden) {
	Class<?> c = extensionClasses.get(name);
	if (c == null || overridden) {
		extensionClasses.put(name, clazz);
	} else if (c != clazz) {
		String duplicateMsg = "Duplicate extension " + type.getName() + " name " + name + " on " + c.getName() + " and " + clazz.getName();
		logger.error(duplicateMsg);
		throw new IllegalStateException(duplicateMsg);
	}
}
```

From the `master` branch, we can see that there are two conditions that will not result in a Duplicate Exception

+ `c == null`: if it's the first time to load the class with the given name
+ `overridden == true`: it depends on `LoadingStrategy`

For all of default loading strategies, The values of `overridden`  are list as follows:

+ [DubboLoadingStrategy](https://github.com/apache/dubbo/blob/master/dubbo-common/src/main/java/org/apache/dubbo/common/extension/DubboLoadingStrategy.java#L33)'s `overridden` is `true`, which is used to resolve `META-INF/dubbo/com.limore.ExtensionConflict`
+ [ServicesLoadingStrategy](https://github.com/apache/dubbo/blob/master/dubbo-common/src/main/java/org/apache/dubbo/common/extension/ServicesLoadingStrategy.java#L33)'s `overridden` is `true`
+ [DubboInternalLoadingStrategy](https://github.com/apache/dubbo/blob/master/dubbo-common/src/main/java/org/apache/dubbo/common/extension/DubboInternalLoadingStrategy.java)'s `overridden` is `false`, which is used to resolve `META-INF/dubbo/internal/com.limore.ExtensionConflict`

We can see the implement class defined in `META-INF/dubbo/internal/com.limore.ExtensionConflict` can be overridden, so the situation you mentioned will never occur.



Also, I have check the [dubbo 2.6.6](https://github.com/apache/dubbo/releases/tag/dubbo-2.6.6), which is released on Feb 28, 2019 (The day after you asked the question)

```java
private void loadClass(Map<String, Class<?>> extensionClasses, java.net.URL resourceURL, Class<?> clazz, String name) throws NoSuchMethodException {
        ......
        if (clazz.isAnnotationPresent(Adaptive.class)) {
            ......
        } else if (isWrapperClass(clazz)) {
            ......
        } else {
            ......
            String[] names = NAME_SEPARATOR.split(name);
            if (names != null && names.length > 0) {
                ......
                for (String n : names) {
                    if (!cachedNames.containsKey(clazz)) {
                        cachedNames.put(clazz, n);
                    }
                    Class<?> c = extensionClasses.get(n);
                  	// don't need to check overridden
                    if (c == null) {
                        extensionClasses.put(n, clazz);
                    } else if (c != clazz) {
                        throw new IllegalStateException("Duplicate extension " + type.getName() + " name " + n + " on " + c.getName() + " and " + clazz.getName());
                    }
                }
            }
        }
    }
```
There are no `overridden` to check, so your issue will occur at that time.

And now, I have fixed this issue in the `master` branch. You can see #7787 , if you're interested.

**Comment by AlbumenJ** [comment]
Fixed in #7787
